### PR TITLE
Adding MachineConfig package

### PR DESCRIPTION
**Pull request by cdvultur**
Adding support for MachineConfig that allows support for changing configuration of a node.

Added suport for 
- WithKernelArguments
- WithExtensions
- WithFips
- WithKernelType

### CODE REVIEW

File: pkg/mco/mc.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the machineconfig is empty")
+
+		builder.errorMsg = "machineconfig 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("machineconfig object #{name} doesn't exist")
```
Comment (trewest) [inline pkg/mco/mc.go:74]:
```suggestion
		return nil, fmt.Errorf("machineconfig object %s doesn't exist", name)
```

File: pkg/mco/mc.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+	if !builder.Exists() {
+		return fmt.Errorf("MachineConfig cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.MachineConfigs().Delete(
+		context.TODO(), builder.Object.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return fmt.Errorf("cannot delete MachineConfig :%w", err)
```
Comment (trewest) [inline pkg/mco/mc.go:115]:
```suggestion
		return fmt.Errorf("cannot delete MachineConfig: %w", err)
```

File: pkg/mco/mc.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+		return false
+	}
+
+	glog.V(100).Infof("Checking if the MachineConfig object %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.MachineConfigs().Get(
+		context.Background(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || k8serrors.IsNotFound(err)
```
Comment (trewest) [inline pkg/mco/mc.go:150]:
```suggestion
	return err == nil || !k8serrors.IsNotFound(err)
```

File: pkg/mco/mc.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
+package mco
```
Comment (trewest) [inline pkg/mco/mc.go:1]: can this file be renamed as machineconfig.go ? I don't want the brevity of the filename to cause folks to easily overlook it.

**Review by trewest — CHANGES_REQUESTED** [review]

File: pkg/mco/mc.go (hunk 3 above)
Comment (cdvultur) [reply pkg/mco/mc.go]: Seems that I forgot to push it. 

Again, BIG Thanks !

File: pkg/mco/mc.go (hunk 2 above)
Comment (cdvultur) [reply pkg/mco/mc.go]: Fixed. Thanks!

File: pkg/mco/mc.go (hunk 1 above)
Comment (cdvultur) [reply pkg/mco/mc.go]: Fixed. Thanks!

**Review by trewest — DISMISSED** [review]
/lgtm

File: pkg/mco/machineconfig.go [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+	if !builder.Exists() {
+		return nil, fmt.Errorf("machineconfig object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Create builds machineconfig in the cluster and stores the created object in struct.
```
Comment (kononovn) [inline pkg/mco/machineconfig.go:82]: Please fix the comment

File: pkg/mco/machineconfig.go [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+
+	var err error
+	builder.Object, err = builder.apiClient.MachineConfigs().Get(
+		context.Background(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithLabel redefines machineconfig definition with the given label.
+func (builder *MCBuilder) WithLabel(key string, value string) *MCBuilder {
```
Comment (kononovn) [inline pkg/mco/machineconfig.go:154]: Please refactor func arguments:
(key,value string)

File: pkg/mco/machineconfig.go [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+				return builder
+			}
+		}
+	}
+
+	return builder
+}
+
+// WithKernelArguments sets the specified KernelArguments to the MachineConfig.
+func (builder *MCBuilder) WithKernelArguments(kernelArgs []string) *MCBuilder {
```
Comment (kononovn) [inline pkg/mco/machineconfig.go:204]: Please validate that kernelArgs is not empty list.

File: pkg/mco/machineconfig.go [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+
+	glog.V(100).Infof("Setting KernelArguments: %v", kernelArgs)
+
+	builder.Definition.Spec.KernelArguments = kernelArgs
+
+	return builder
+}
+
+// WithExtensions sets the specified Extensions to the MachineConfig.
+func (builder *MCBuilder) WithExtensions(extensions []string) *MCBuilder {
```
Comment (kononovn) [inline pkg/mco/machineconfig.go:217]: Please validate that extensions is not empty list.

File: pkg/mco/machineconfig.go [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+
+	glog.V(100).Infof("Setting FIPS: %v", fips)
+
+	builder.Definition.Spec.FIPS = fips
+
+	return builder
+}
+
+// WithKernelType sets the specified kernelType to the MachineConfig.
+func (builder *MCBuilder) WithKernelType(kernelType string) *MCBuilder {
```
Comment (kononovn) [inline pkg/mco/machineconfig.go:243]: Please check that kernelType is not empty sting.

**Review by kononovn — CHANGES_REQUESTED** [review]
Please fix few comments above.

File: pkg/mco/machineconfig.go [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+
+	glog.V(100).Infof("Setting Extensions: %v", extensions)
+
+	builder.Definition.Spec.Extensions = extensions
+
+	return builder
+}
+
+// WithFips sets the specified FIPS value to the MachineConfig.
+func (builder *MCBuilder) WithFips(fips bool) *MCBuilder {
```
Comment (trewest) [inline pkg/mco/machineconfig.go:230]: nit: WithFIPS

File: pkg/mco/machineconfig.go (hunk 5 above)
Comment (cdvultur) [reply pkg/mco/machineconfig.go]: Updated comment.

Thanks!

File: pkg/mco/machineconfig.go (hunk 6 above)
Comment (cdvultur) [reply pkg/mco/machineconfig.go]: Updated. Thanks!

File: pkg/mco/machineconfig.go (hunk 9 above)
Comment (cdvultur) [reply pkg/mco/machineconfig.go:259]: Added empty string check.

Thanks!

File: pkg/mco/machineconfig.go (hunk 10 above)
Comment (cdvultur) [reply pkg/mco/machineconfig.go]: Update it to be coherent with the K8s definition.

Thanks!

**Review by kononovn — APPROVED** [review]
lgtm

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**